[2025-04-16T15:27:07.225+0300] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-04-16T15:27:07.237+0300] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: executing_Python_Branching_tasks_with_Backfill.task_choice scheduled__2025-04-15T00:00:00+00:00 [queued]>
[2025-04-16T15:27:07.243+0300] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: executing_Python_Branching_tasks_with_Backfill.task_choice scheduled__2025-04-15T00:00:00+00:00 [queued]>
[2025-04-16T15:27:07.244+0300] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-04-16T15:27:07.264+0300] {taskinstance.py:2890} INFO - Executing <Task(BranchPythonOperator): task_choice> on 2025-04-15 00:00:00+00:00
[2025-04-16T15:27:07.267+0300] {standard_task_runner.py:72} INFO - Started process 144734 to run task
[2025-04-16T15:27:07.273+0300] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'executing_Python_Branching_tasks_with_Backfill', 'task_choice', 'scheduled__2025-04-15T00:00:00+00:00', '--job-id', '143', '--raw', '--subdir', 'DAGS_FOLDER/execute_cron_catchup_backfill.py', '--cfg-path', '/tmp/tmpao8jar0c']
[2025-04-16T15:27:07.275+0300] {standard_task_runner.py:105} INFO - Job 143: Subtask task_choice
[2025-04-16T15:27:07.329+0300] {task_command.py:467} INFO - Running <TaskInstance: executing_Python_Branching_tasks_with_Backfill.task_choice scheduled__2025-04-15T00:00:00+00:00 [running]> on host DESKTOP-1SBO1JU.
[2025-04-16T15:27:07.411+0300] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='tasneemahmed' AIRFLOW_CTX_DAG_ID='executing_Python_Branching_tasks_with_Backfill' AIRFLOW_CTX_TASK_ID='task_choice' AIRFLOW_CTX_EXECUTION_DATE='2025-04-15T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-04-15T00:00:00+00:00'
[2025-04-16T15:27:07.411+0300] {taskinstance.py:732} INFO - ::endgroup::
[2025-04-16T15:27:07.427+0300] {baseoperator.py:423} WARNING - BranchPythonOperator.execute cannot be called outside TaskInstance!
[2025-04-16T15:27:07.427+0300] {python.py:240} INFO - Done. Returned value was: False
[2025-04-16T15:27:07.428+0300] {branch.py:38} INFO - Branch into False
[2025-04-16T15:27:07.432+0300] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 274, in execute
    return self.do_branch(context, super().execute(context))
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/operators/branch.py", line 40, in do_branch
    self.skip_all_except(context["ti"], branch_task_ids)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/models/skipmixin.py", line 192, in skip_all_except
    SkipMixin._skip_all_except(ti=ti, branch_task_ids=branch_task_ids)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/api_internal/internal_api_call.py", line 166, in wrapper
    return func(*args, **kwargs)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/models/skipmixin.py", line 221, in _skip_all_except
    raise AirflowException(
airflow.exceptions.AirflowException: 'branch_task_ids' expected all task IDs are strings. Invalid tasks found: {(False, 'bool')}.
[2025-04-16T15:27:07.437+0300] {taskinstance.py:1226} INFO - Marking task as FAILED. dag_id=executing_Python_Branching_tasks_with_Backfill, task_id=task_choice, run_id=scheduled__2025-04-15T00:00:00+00:00, execution_date=20250415T000000, start_date=20250416T122707, end_date=20250416T122707
[2025-04-16T15:27:07.461+0300] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-04-16T15:27:07.462+0300] {standard_task_runner.py:124} ERROR - Failed to execute job 143 for task task_choice ('branch_task_ids' expected all task IDs are strings. Invalid tasks found: {(False, 'bool')}.; 144734)
Traceback (most recent call last):
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 274, in execute
    return self.do_branch(context, super().execute(context))
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/operators/branch.py", line 40, in do_branch
    self.skip_all_except(context["ti"], branch_task_ids)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/models/skipmixin.py", line 192, in skip_all_except
    SkipMixin._skip_all_except(ti=ti, branch_task_ids=branch_task_ids)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/api_internal/internal_api_call.py", line 166, in wrapper
    return func(*args, **kwargs)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/tasneemahmed24/.local/lib/python3.10/site-packages/airflow/models/skipmixin.py", line 221, in _skip_all_except
    raise AirflowException(
airflow.exceptions.AirflowException: 'branch_task_ids' expected all task IDs are strings. Invalid tasks found: {(False, 'bool')}.
[2025-04-16T15:27:07.499+0300] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-04-16T15:27:07.519+0300] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-04-16T15:27:07.528+0300] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-04-16T16:46:12.393+0300] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-04-16T16:46:12.400+0300] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: executing_Python_Branching_tasks_with_Backfill.task_choice scheduled__2025-04-15T00:00:00+00:00 [queued]>
[2025-04-16T16:46:12.405+0300] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: executing_Python_Branching_tasks_with_Backfill.task_choice scheduled__2025-04-15T00:00:00+00:00 [queued]>
[2025-04-16T16:46:12.405+0300] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-04-16T16:46:12.416+0300] {taskinstance.py:2890} INFO - Executing <Task(PythonOperator): task_choice> on 2025-04-15 00:00:00+00:00
[2025-04-16T16:46:12.418+0300] {standard_task_runner.py:72} INFO - Started process 177799 to run task
[2025-04-16T16:46:12.423+0300] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'executing_Python_Branching_tasks_with_Backfill', 'task_choice', 'scheduled__2025-04-15T00:00:00+00:00', '--job-id', '151', '--raw', '--subdir', 'DAGS_FOLDER/execute_cron_catchup_backfill.py', '--cfg-path', '/tmp/tmp1idao052']
[2025-04-16T16:46:12.424+0300] {standard_task_runner.py:105} INFO - Job 151: Subtask task_choice
[2025-04-16T16:46:12.462+0300] {task_command.py:467} INFO - Running <TaskInstance: executing_Python_Branching_tasks_with_Backfill.task_choice scheduled__2025-04-15T00:00:00+00:00 [running]> on host DESKTOP-1SBO1JU.
[2025-04-16T16:46:12.509+0300] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='tasneemahmed' AIRFLOW_CTX_DAG_ID='executing_Python_Branching_tasks_with_Backfill' AIRFLOW_CTX_TASK_ID='task_choice' AIRFLOW_CTX_EXECUTION_DATE='2025-04-15T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-04-15T00:00:00+00:00'
[2025-04-16T16:46:12.510+0300] {taskinstance.py:732} INFO - ::endgroup::
[2025-04-16T16:46:12.518+0300] {python.py:240} INFO - Done. Returned value was: True
[2025-04-16T16:46:12.530+0300] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-04-16T16:46:12.530+0300] {taskinstance.py:353} INFO - Marking task as SUCCESS. dag_id=executing_Python_Branching_tasks_with_Backfill, task_id=task_choice, run_id=scheduled__2025-04-15T00:00:00+00:00, execution_date=20250415T000000, start_date=20250416T134612, end_date=20250416T134612
[2025-04-16T16:46:12.564+0300] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-04-16T16:46:12.575+0300] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-04-16T16:46:12.575+0300] {local_task_job_runner.py:245} INFO - ::endgroup::
